src/common/notifications.cpp

```diff
@@ -315,3 +319,99 @@ void notifications::show_toast_with_activations(std::wstring message, std::wstri
 
     notifier.Show(notification);
 }
+
+void notifications::show_toast_with_progress_bar(std::wstring message, const std::wstring& version, toast_params params)
+{
+    // DO NOT LOCALIZE any string in this function, because they're XML tags and a subject to
+    // https://docs.microsoft.com/en-us/windows/uwp/design/shell/tiles-and-notifications/toast-xml-schema
+
+    if (!params.progress.has_value())
+    {
+        return;
+    }
+
+    std::wstring toast_xml;
+    toast_xml.reserve(2048);
+    std::wstring title{ L"PowerToys" };
+    if (winstore::running_as_packaged())
+    {
+        title += L" (Experimental)";
+    }
+
+    toast_xml += LR"(<?xml version="1.0"?><toast><visual><binding template="ToastGeneric"><text>)";
+    toast_xml += title;
+    toast_xml += L"</text><text>";
+    toast_xml += message;
+    toast_xml += L"</text>";
+    toast_xml += LR"(<progress title=")";
+    toast_xml += localized_strings::PT_UPDATE;
+    toast_xml += L" ";
+    toast_xml += version;
+    toast_xml += LR"(" value="{progressValue}" valueStringOverride="{progressValueString}" status="{progressStatus}"/>)";
+    toast_xml += L"</binding></visual></toast>";
+
+    XmlDocument toast_xml_doc;
+    xml_escape(toast_xml);
+    toast_xml_doc.LoadXml(toast_xml);
+    ToastNotification notification{ toast_xml_doc };
+
+    float progress = params.progress.value();
```

Comment:
Let's [clamp](https://en.cppreference.com/w/cpp/algorithm/clamp) the value in [0,1].